packages/core/minikit.ts

```diff
@@ -94,6 +100,13 @@ export class MiniKit {
       ) => {
         if (payload.status === "success") {
           MiniKit.walletAddress = payload.address;
+          getUserProfile(payload.address).then((queryResponse) => {
```

Comment:
@0xPenryn thanks for this utility, out of curisiosity why is this call asynchronous?

What happens when I use walletAuth is that I get the MiniKit.walletAddress instantly after having called walletAuth, but the additional information (userProfile) is added a bit later because this function returns the originalHandler before getUserProfile has been executed.

It's not really practictal in my react environment, I wonder how you manage this on your side

Reply:
Hey we merged an improvement haven't released it yet https://github.com/worldcoin/minikit-js/pull/122